frontend/src/components/Form.tsx

```diff
@@ -0,0 +1,95 @@
+import { useState } from 'react'
+import { FormField } from './FormField'
+
+export interface FormData {
+  name: string
+  age: string
+  favoriteColor: string
+  preferredContactMethod: string
+}
+
+export const Form = () => {
+  const [formData, setFormData] = useState({
+    name: '',
+    age: '',
+    favoriteColor: '',
+    preferredContactMethod: 'Email',
+  })
+  const contactMethods = ['Email', 'Phone']
+
+  const handleSubmitForm = async (e: React.FormEvent<HTMLFormElement>) => {
+    e.preventDefault()
+
+    if (formData.name === '' || formData.age === '' || formData.favoriteColor === '') {
```

Comment:
Nit: It's not a problem per-se, but the behavior is tightly dependent on a default value of `preferredContactMethod` in a form. If the `preferredContactMethod` default value was changed to empty (perhaps to make sure user selects something explicitly), this code would suddenly become bugged as it would pass a request without a form. It's safer to check for all fields, if they are all required.